custom_components/tahoma/water_heater.py

```diff
@@ -3,19 +3,32 @@
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from pyoverkiz.enums import UIWidget
+from pyoverkiz.enums import OverkizState, UIWidget
 
 from .const import DOMAIN
 from .water_heater_devices.domestic_hot_water_production import (
     DomesticHotWaterProduction,
 )
+from .water_heater_devices.domestic_hot_water_production_sauter_guelma import (
+    DomesticHotWaterProductionSauterGUELMA,
+)
 from .water_heater_devices.hitachi_dhw import HitachiDHW
 
 TYPE = {
     UIWidget.DOMESTIC_HOT_WATER_PRODUCTION: DomesticHotWaterProduction,
     UIWidget.HITACHI_DHW: HitachiDHW,
+    UIWidget.DOMESTIC_HOT_WATER_PRODUCTION
+    + "SauterGUELMA": DomesticHotWaterProductionSauterGUELMA
```

Comment:
I am not a huge fan of this + the device_type array. I think it will break other implementations that don't have a manufacturer name / core_name, and brings a lot of extra complexity.

Could we perhaps just look from the class itself which features we need to enable? I am not sure if these appliances are very different?

Reply:
I agree this is not ideal. I tried several implementations including having the DomesticHotWaterProduction class handle the device.
The major drawback is the addition of extra complexity while handling property 'is_away_mode_on' or callable to set the away mode of the device because my device (Sauter GUELMA) operating mode is different from Atlantic DHW.


Since the ui.widget state is identical to Genuine Atlantic DHW, using Manufacturer and Model state was my only way to discriminate the device to instantiate the right class.
I'm just beginning my journey with HA so I'll be glad to ear from you on how I can improve.
Best,